x-pack/plugin/sql/qa/src/main/java/org/elasticsearch/xpack/sql/qa/rest/BaseRestSqlTestCase.java

```diff
@@ -19,10 +18,119 @@
 
 import java.io.IOException;
 import java.io.InputStream;
+import java.util.Locale;
 import java.util.Map;
 
+import static org.elasticsearch.xpack.sql.proto.Protocol.BINARY_FORMAT_NAME;
+import static org.elasticsearch.xpack.sql.proto.Protocol.CLIENT_ID_NAME;
+import static org.elasticsearch.xpack.sql.proto.Protocol.VERSION_NAME;
+import static org.elasticsearch.xpack.sql.proto.Protocol.COLUMNAR_NAME;
+import static org.elasticsearch.xpack.sql.proto.Protocol.CURSOR_NAME;
+import static org.elasticsearch.xpack.sql.proto.Protocol.FETCH_SIZE_NAME;
+import static org.elasticsearch.xpack.sql.proto.Protocol.FILTER_NAME;
+import static org.elasticsearch.xpack.sql.proto.Protocol.MODE_NAME;
+import static org.elasticsearch.xpack.sql.proto.Protocol.PARAMS_NAME;
+import static org.elasticsearch.xpack.sql.proto.Protocol.QUERY_NAME;
+import static org.elasticsearch.xpack.sql.proto.Protocol.TIME_ZONE_NAME;
+
 public abstract class BaseRestSqlTestCase extends ESRestTestCase {
-    
+
+    public static class RequestObjectBuilder {
+        private StringBuilder request;
+        private final boolean isQuery;
+
+        private RequestObjectBuilder(String init, boolean isQuery) {
+            request = new StringBuilder(init);
+            this.isQuery = isQuery;
+        }
+
+        public static RequestObjectBuilder query(String query) {
+            return new RequestObjectBuilder(field(QUERY_NAME, query).substring(1), true);
+        }
+
+        public static RequestObjectBuilder cursor(String cursor) {
+            return new RequestObjectBuilder(field(CURSOR_NAME, cursor).substring(1), false);
+        }
+
+        public RequestObjectBuilder version(String version) {
+            request.append(field(VERSION_NAME, version));
+            return this;
+        }
+
+        public RequestObjectBuilder mode(String mode) {
```

Comment:
I would add a method that accepts Mode directly so callers don't have to keep calling `mode(someMode.toString())`

Reply:
Good idea, thanks.